app/javascript/controllers/menu_button_controller.ts

```diff
@@ -17,6 +16,10 @@ export class MenuButtonController extends ApplicationController {
     this.#teardown?.();
   }
 
+  private get isOpen() {
+    return (this.element as HTMLElement).classList.contains('open');
```

Comment:
Un truc m'échappe un peu : dans les funcitons open/close on toggle la classe sur  le parent, par ex :  `this.menuTarget.parentElement?.classList.add('open');`

Est-ce qu'il ne faudrait pas faire l'assertion sur le parent aussi ?

Reply:
`element` c'est le `parentElement` du `menuTarget`. Je pense qu'on pourrait remplacer `this.menuTarget.parentElement` par `this.element`. On peut faire un refacto dans une autre PR pour clarifier tout ça et aussi wrapper dans un component ruby.

Reply:
c'est vraiment top. je ferme la PR ou tu avais réagis a mon hack la dessus. well done !